File: AbetOrder/OpenDataAccess/OpenDataAccess.Service/ComponentType.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenDataAccess.Service
{
    public class ComponentType
    {
        public static ConcurrentBag<Type> TypeList = new ConcurrentBag<Type>();

        public static Type GetComponentType(string name)
        {
            return TypeList.Where(where => where.Name.Trim().ToLower() == name.Trim().ToLower()).FirstOrDefault();
        }

        public static void SetComponentType(Type type)
        {
            if (TypeList.Where(exists => exists.Name == type.Name).FirstOrDefault() == null)
            {
                TypeList.Add(type);
            }
        }
    }
}

File: JavaWebFramework/src/OpenDataAccess/Service/ComponentType.java
package OpenDataAccess.Service;

import OpenDataAccess.Utility.Common;

import java.util.ArrayList;
import java.util.List;

public class ComponentType {
    public static List<Class<?>> TypeList = new ArrayList<>();

    public static Class<?> GetComponentType(String name) {
        for (int i = 0; i < TypeList.size(); i++) {
            if (TypeList.get(i).getName().equals(name)) return TypeList.get(i);
        }
        return null;
    }

    public static void SetComponentType(Class<?> type) {
        if (!Common.Exists(TypeList, e -> e.getName().equals(type.getName()))) TypeList.add(type);
    }
}
